app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
```

Comment:
I agree, perhaps this would indicate it slightly better

```ruby
subnet = [self.subnet, self.subnet6].compact
return subnet.vlanid unless subnet.nil?
```

but maybe just a comment explaining what you described here would do :-) Ultimate solution would be to display a warning if user uses two subnets with different vlan tag. But I suppose that's out of scope.